eventbridge/src/main/scala/uk/gov/nationalarchives/DAEventBridgeClient.scala

```diff
@@ -0,0 +1,67 @@
+package uk.gov.nationalarchives
+
+import cats.effect.Async
+import io.circe.syntax.EncoderOps
+import io.circe.{Encoder, Json, Printer}
+import org.typelevel.log4cats.SelfAwareStructuredLogger
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.eventbridge.EventBridgeAsyncClient
+import software.amazon.awssdk.services.eventbridge.model.{PutEventsRequest, PutEventsRequestEntry, PutEventsResponse}
+
+/** An EventBridgeAsyncClient client. It is written generically so can be used for any effect which has an Async
+  * instance. Requires an implicit instance of cats Async which is used to convert CompletableFuture to F
+  *
+  * @param asyncClient
+  *   An AWS EventBridge Async client
+  * @tparam F
+  *   Type of the effect
+  */
+class DAEventBridgeClient[F[_]: Async](asyncClient: EventBridgeAsyncClient) {
+
+  /** Sends an event to EventBridge. detail will be deserialised to a json string.
+    * @param sourceId
+    *   The source ID to send to EventBridge
+    * @param detailType
+    *   The detail type to send to EventBridge
+    * @param detail
+    *   A case class of type T. This will be deserialised to a json string
+    * @param enc
+    *   The circe encoder to do the deserialisation
+    * @tparam T
+    *   The type of the detail class
+    * @return
+    */
+  def publishEventToEventBridge[T](sourceId: String, detailType: String, detail: T)(implicit
```

Comment:
1. You added `[T <: Product]` in the docs but just `[T]` here
2. Also, I was thinking, since `detail` will be deserialised to a JSON, do you think it's worth changing it to `[T < Product with Serializable]`?

Reply:
I changed it to just `T` after I wrote that. I changed it because with `[T <: Product]` and `[T < Product with Serializable]` you can't pass in a trait which is something we might want to do so I left it as `T` It doesn't really metter what you pass as long as there's a valid circe encoder for it so the type bounds aren't needed.
I've updated the docs.